### PR TITLE
docs: add documentation for ACC agent pools running Ubuntu 18.04

### PR DESCRIPTION
**Pull request by dmitsh**
**Reason for Change**:

Added documentation for supporting Ubuntu 18.04 on DC-series VMs (ACC)
Cleaned up driver installation steps

**Issue Fixed**:

Related to #317

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: docs/topics/sgx.md [hunk 1]
```diff
@@ -3,15 +3,28 @@
 [Intel&reg; Secure Guard Extension](https://software.intel.com/en-us/sgx) (Intel&reg; SGX) is an architecture extension designed to increase the security of application code and data.
 Developers may choose [Intel&reg; SGX SDK](https://software.intel.com/en-us/sgx-sdk) or [Open Enclave SDK](https://github.com/Microsoft/openenclave/) to create applications that leverage this technology.
 
-Azure supports provisioning of SGX-enabled VMs under the umbrella of Azure Confidential Compute (ACC). You can create a Kubernetes cluster with one or multiple agent pool(s) running ACC VMs by specifying VM size `Standard_DC*` and distro `acc-16.04`.
+Azure supports provisioning of SGX-enabled VMs under the umbrella of Azure Confidential Compute (ACC). You can create a Kubernetes cluster with one or multiple agent pool(s) running ACC VMs by specifying [DC-series](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/sizes-general#dc-series) VM size and the supported distro from the table below.
```
Comment (mboersma) [inline docs/topics/sgx.md]: Could you add `a` to "by specifying a [DC-series](https://..." and change the second `and the` to `with a`?
```
You can create a Kubernetes cluster with one or multiple agent pool(s) running ACC VMs by specifying 
a [DC-series](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/sizes-general#dc-series)
VM size with a supported distro from the table below.
```

File: docs/topics/sgx.md [hunk 2]
```diff
@@ -3,15 +3,28 @@
...
+
+| OS           | distro      | Notes |
+| ------------ | ----------- |-------|
+| Ubuntu 16.04 | `acc-16.04` | specially built image with UEFI BIOS support
+| Ubuntu 18.04 | `aks-18.04` | standard Canonical Ubuntu image
+
+The following example is a fragment of api-model file declaring two ACC agent pools, one running `Ubuntu 16.04` image on `2 vCPU` nodes, and another running `Ubuntu 18.04` image on `4 vCPU` nodes:
```
Comment (mboersma) [inline docs/topics/sgx.md]: Could you change the first part of the sentence like this:
`The following example is a fragment of a cluster definition (apimodel) file declaring...`

File: parts/k8s/kubernetescustomscript.sh [hunk 3]
```diff
@@ -86,9 +86,7 @@ installKubeletAndKubectl
 ensureRPC
 createKubeManifestDir
 if [[ "${SGX_NODE}" = true ]]; then
-    if $FULL_INSTALL_REQUIRED; then
-        installSGXDrivers
-    fi
+    installSGXDrivers
```
Comment (mboersma) [inline parts/k8s/kubernetescustomscript.sh]: This change means `installSGXDrivers` is called always during provisioning, even if the drivers were already installed on the VHD image. That seems like a de-optimization. Does this fix something?

Comment (dmitsh) [reply parts/k8s/kubernetescustomscript.sh]: The `FULL_INSTALL_REQUIRED` does not apply to SGX driver installation. It was copy-pasted by mistake. Cleaning it up.

File: docs/topics/sgx.md [hunk 4]
```diff
@@ -3,15 +3,28 @@
...
 Developers may choose [Intel&reg; SGX SDK](https://software.intel.com/en-us/sgx-sdk) or [Open Enclave SDK](https://github.com/Microsoft/openenclave/) to create applications that leverage this technology.
 
-Azure supports provisioning of SGX-enabled VMs under the umbrella of Azure Confidential Compute (ACC). You can create a Kubernetes cluster with one or multiple agent pool(s) running ACC VMs by specifying VM size `Standard_DC*` and distro `acc-16.04`.
+Azure supports provisioning of SGX-enabled VMs under the umbrella of Azure Confidential Compute (ACC). You can create a Kubernetes cluster with one or multiple agent pool(s) running ACC VMs by specifying a [DC-series](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/sizes-general#dc-series) VM size with a supported distro from the table below.
+
+| OS           | distro      | Notes |
+| ------------ | ----------- |-------|
+| Ubuntu 16.04 | `acc-16.04` | specially built image with UEFI BIOS support
+| Ubuntu 18.04 | `aks-18.04` | standard Canonical Ubuntu image
```
Comment (CecileRobertMichon) [inline docs/topics/sgx.md]: `aks-18.04` isn't the "standard Canonical Ubuntu image", it's a distro curated by AKS Engine with pre-installed components.

File: docs/topics/sgx.md [hunk 5]
```diff
@@ -3,15 +3,28 @@
...
+
+The following example is a fragment of a cluster definition (apimodel) file declaring two ACC agent pools, one running `Ubuntu 16.04` image on `2 vCPU` nodes, and another running `Ubuntu 18.04` image on `4 vCPU` nodes:
+
```
Comment (jackfrancis) [inline docs/topics/sgx.md]: You'll probably want to comment that this cluster will have to be kubenet-backed

Comment (dmitsh) [reply docs/topics/sgx.md]: I was hoping that this issue will be resolved at some point....

File: docs/topics/sgx.md (hunk 4 above)
Comment (dmitsh) [reply docs/topics/sgx.md]: Thank you @CecileRobertMichon . Right, I noticed `microsoft-aks` image publisher.
I will replace this line with `standard AKS image for Ubuntu 18.04`.
Would that be more descriptive?

**Pull request merged**